packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -570,6 +570,15 @@ export class SourceFile {
         try {
             //TODO: this isnt ideal because it re-reads the file for each cell which is unnecessary
             source = getIPythonCells(this.fileSystem, this.getRealUri(), this._console)?.[cellIndex]?.source;
+            // Make sure we don't return any non-python cells
+            if (
+                source &&
+                ['%sql', '%%sql', '%sh', '%pip', '%run', '%fs', '%load', '%matplotlib', '%who', '%env'].includes(
```

Comment:
i don't think this is the correct spot for this because this method is only called when running basedpyright from the CLI. when using the language server, the file contents get passed to the `SourceFile` object by writing to `this._writableData.clientDocumentContents`

Reply:
Ok, is there a method or so where this filtering would make sense? How do I test the language server?